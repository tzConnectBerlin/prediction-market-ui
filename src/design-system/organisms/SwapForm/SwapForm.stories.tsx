import * as React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { SwapForm, SwapFormProps } from './SwapForm';

export default {
  title: 'Organism/SwapForm',
  component: SwapForm,
} as Meta;

const Template: Story<SwapFormProps> = (args) => <SwapForm {...args} />;

export const Mint = Template.bind({});
Mint.args = {
  tokenName: 'PMM',
  title: 'Mint',
  handleSubmit: () => console.log('submit'),
};
