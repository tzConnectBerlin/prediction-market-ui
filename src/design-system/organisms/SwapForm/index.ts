export { SwapForm, SwapForm as default } from './SwapForm';
export type { SwapFormProps } from './SwapForm';
