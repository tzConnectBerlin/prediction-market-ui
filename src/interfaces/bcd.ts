export interface SimilarContract {
  added?: number;
  address: string;
  alias?: string;
  annotations?: string[];
  balance?: number;
  count?: number;
  delegate?: string;
  delegate_alias?: string;
  entrypoints?: string[];
  fail_strings?: string[];
  found_by?: string;
  hardcoded?: string[];
  hash: string;
  id?: string;
  language?: string;
  last_action?: string;
  level?: number;
  manager?: string;
  migrations_count?: number;
  network?: string;
  project_id?: string;
  removed?: number;
  slug?: string;
  subscription?: Subscription;
  tags?: string[];
  timestamp?: string;
  tokens?: TokenBalance[];
  total_subscribed?: number;
  total_withdrawn?: number;
  tx_count?: number;
  verification_source?: string;
  verified?: boolean;
}

export interface Subscription {
  address?: string;
  alias?: string;
  network?: string;
  sentry_dsn?: string;
  sentry_enabled?: boolean;
  subscribed_at?: string;
  watch_calls?: boolean;
  watch_deployments?: boolean;
  watch_errors?: boolean;
  watch_mempool?: boolean;
  watch_migrations?: boolean;
  watch_same?: boolean;
  watch_similar?: boolean;
}

export interface TokenBalance {
  balance?: number;
  contract?: string;
  decimals?: number;
  name?: string;
  symbol?: string;
  token_id?: number;
}

export interface SimilarContractResponse {
  contracts?: SimilarContract[];
  count?: number;
}
