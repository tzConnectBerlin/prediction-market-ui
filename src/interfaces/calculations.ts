export interface ClosePositionReturn {
  aToSwap: number;
  aLeft: number;
  bReceived: number;
  aToSwapWithSlippage: number;
}

export interface ClosePositionBothReturn
  extends Omit<ClosePositionReturn, 'bReceived' | 'aToSwapWithSlippage'> {
  bHeld: number;
}

export interface BuyPosition {
  price: number;
  swap: number;
  quantity: number;
}
