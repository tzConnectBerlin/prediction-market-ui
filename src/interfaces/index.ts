export * from './wallet';
export * from './router';
