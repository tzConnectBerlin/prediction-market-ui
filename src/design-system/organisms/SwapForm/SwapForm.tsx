import React from 'react';
import * as Yup from 'yup';
import { Field, Form, Formik, FormikHelpers } from 'formik';
import { useTranslation } from 'react-i18next';
import { Grid, Theme, useTheme } from '@material-ui/core';
import { SxProps } from '@material-ui/system';
import { FormikTextField } from '../../molecules/FormikTextField';
import { Typography } from '../../atoms/Typography';
import { CustomButton } from '../../atoms/Button';
import { PositionItem, PositionSummary } from '../SubmitBidCard/PositionSummary';
import { getNoTokenId, getTokenQuantityById, getYesTokenId } from '../../../utils/misc';
import { Token, TokenType } from '../../../interfaces';
import { roundToTwo, tokenDivideDown, tokenMultiplyUp } from '../../../utils/math';
import { swapTokenCalculations, tokenAmountAfterSwap } from '../../../contracts/MarketCalculations';
import { IconTooltip } from '../../atoms/IconTooltip';

const endAdornmentStyles: SxProps<Theme> = { whiteSpace: 'nowrap' };
const TokenPriceDefault = {
  yes: 0,
  no: 0,
};

export type SwapFormValues = {
  yesToken: string | number;
  noToken: string | number;
  swapTokenType: TokenType;
};
export interface SwapFormProps {
  title: string;
  /**
   * Callback to get the form values
   */
  handleSubmit: (
    values: SwapFormValues,
    formikHelpers: FormikHelpers<SwapFormValues>,
  ) => void | Promise<void>;
  /**
   * Initial values to use when initializing the form. Default is 0.
   */
  initialValues?: Omit<SwapFormValues, 'swapTokenType'>;
  /**
   * Is wallet connected
   */
  connected?: boolean;
  /**
   * TokenName to display
   */
  tokenName?: string;
  /**
   * Token Price
   */
  tokenPrice?: {
    yes: number;
    no: number;
  };
  /**
   * Market Id
   */
  marketId: string;
  /**
   * swap token type
   */
  swapTokenType: TokenType;
  /**
   * User token values
   */
  userTokens?: Token[];
  /**
   * User Balance value
   */
  userBalance?: number;
  /**
   * Pool token values
   */
  poolTokens?: Token[];
  /**
   * Slippage
   */
  slippage?: number;
}

export const SwapForm: React.FC<SwapFormProps> = ({
  title,
  handleSubmit,
  initialValues,
  connected,
  tokenName,
  marketId,
  userTokens,
  tokenPrice = TokenPriceDefault,
  swapTokenType,
  userBalance,
  poolTokens,
  slippage = 0,
}) => {
  const theme = useTheme();
  const { t } = useTranslation('common');
  const yesTokenId = React.useMemo(() => getYesTokenId(marketId), [marketId]);
  const noTokenId = React.useMemo(() => getNoTokenId(marketId), [marketId]);
  const [expectedSwap, setExpectedSwap] = React.useState<PositionItem[]>([]);
  const [expectedAdjustedPosition, setexpectedAdjustedPosition] = React.useState<PositionItem[]>(
    [],
  );
  const [userAmounts, setUserAmounts] = React.useState({
    yesToken: 0,
    noToken: 0,
  });
  const [pools, setPools] = React.useState({
    yesPool: 0,
    noPool: 0,
  });
  const [swapToken, otherToken] = TokenType.yes === swapTokenType ? ['Yes', 'No'] : ['No', 'Yes'];
  const PositionDescription = () => (
    <>
      {t('expectedAdjustedPosition')}
      <IconTooltip
        description={t('swapDescription')}
        placement="bottom-start"
        maxWidth={theme.spacing(31)}
      />
    </>
  );
  React.useEffect(() => {
    if (poolTokens) {
      const yesPool = getTokenQuantityById(poolTokens, yesTokenId);
      const noPool = getTokenQuantityById(poolTokens, noTokenId);
      setPools({
        yesPool,
        noPool,
      });
    }
    if (userTokens) {
      const yesToken = getTokenQuantityById(userTokens, yesTokenId);
      const noToken = getTokenQuantityById(userTokens, noTokenId);
      setUserAmounts({
        noToken,
        yesToken,
      });
    }
  }, [poolTokens, userTokens, yesTokenId, noTokenId]);

  const validationSchema = React.useMemo(() => {
    if (swapTokenType === TokenType.yes) {
      if (connected && userBalance) {
        return Yup.object({
          yesToken: Yup.number()
            .min(0.000001, `${t('minSwap')} 0.000001`)
            .max(userBalance, t('insufficientBalance'))
            .required(t('required')),
        });
      }
    } else if (connected && userBalance) {
      return Yup.object({
        noToken: Yup.number()
          .min(0.000001, `${t('minSwap')} 0.000001`)
          .max(userBalance, t('insufficientBalance'))
          .required(t('required')),
      });
    }
    return Yup.object({
      yesToken: Yup.number().max(0, t('insufficientBalance')),
      noToken: Yup.number().max(0, t('insufficientBalance')),
    });
  }, [userAmounts, connected, userBalance]);

  const handleChange = React.useCallback(
    (e: any) => {
      if (Number.parseFloat(e.target.value) > 0) {
        const [aPool, bPool] =
          TokenType.yes === swapTokenType
            ? [pools.yesPool, pools.noPool]
            : [pools.noPool, pools.yesPool];
        const quantity = tokenMultiplyUp(e.target.value);
        const { swapOutput, exchangeRate } = swapTokenCalculations(
          quantity,
          aPool,
          bPool,
          slippage,
        );

        const newExpectedSwap: PositionItem[] = [
          {
            label: t('expectedRate'),
            value: `1 ${t(swapToken)} = ${roundToTwo(exchangeRate)} ${t(otherToken)}`,
          },
          {
            label: t('output'),
            value: `${roundToTwo(tokenDivideDown(swapOutput))} ${t(otherToken)}`,
          },
        ];
        setExpectedSwap(newExpectedSwap);

        if (connected) {
          const [newYes, newNo, yesOpt, noOpt] =
            TokenType.yes === swapTokenType
              ? [quantity, swapOutput, '-', '+']
              : [swapOutput, quantity, '+', '-'];

          const { totalYes, totalNo } = tokenAmountAfterSwap(
            userAmounts.yesToken,
            newYes,
            userAmounts.noToken,
            newNo,
            swapTokenType,
          );

          const newTotalValue = tokenPrice.yes * totalYes + tokenPrice.no * totalNo;

          const newPosition: PositionItem[] = [
            {
              label: `${t(TokenType.yes)} ${t('tokens')}`,
              value: `${roundToTwo(tokenDivideDown(userAmounts.yesToken))} (${yesOpt}${roundToTwo(
                tokenDivideDown(newYes),
              )})`,
            },
            {
              label: `${t(TokenType.no)} ${t('tokens')}`,
              value: `${roundToTwo(tokenDivideDown(userAmounts.noToken))} (${noOpt}${roundToTwo(
                tokenDivideDown(newNo),
              )})`,
            },
            {
              label: t('totalValue'),
              value: `${roundToTwo(tokenDivideDown(newTotalValue))} ${tokenName}`,
            },
          ];
          setexpectedAdjustedPosition(newPosition);
        }
      }
    },
    [connected, pools.yesPool, pools.noPool, userAmounts.noToken, userAmounts.yesToken, t],
  );

  const initialFormValues: SwapFormValues = initialValues
    ? {
        ...initialValues,
        swapTokenType,
      }
    : {
        yesToken: '',
        noToken: '',
        swapTokenType,
      };
  return (
    <Formik
      onSubmit={handleSubmit}
      validationSchema={validationSchema}
      initialValues={initialFormValues}
      enableReinitialize
    >
      {({ isValid }) => (
        <Form>
          <Grid
            container
            spacing={3}
            direction="column"
            alignContent="flex-start"
            justifyContent="center"
          >
            <Grid item width="100%">
              {swapTokenType === TokenType.yes ? (
                <Field
                  component={FormikTextField}
                  label={t('amount')}
                  name="yesToken"
                  type="number"
                  pattern="[0-9]*"
                  fullWidth
                  placeholder={t('inputFieldPlaceholder')}
                  handleChange={handleChange}
                  InputProps={
                    tokenName
                      ? {
                          endAdornment: (
                            <Typography
                              color="text.secondary"
                              component="span"
                              sx={endAdornmentStyles}
                            >
                              {t('yesTokens')}
                            </Typography>
                          ),
                        }
                      : undefined
                  }
                  required
                />
              ) : (
                <>
                  <Field
                    component={FormikTextField}
                    label={t('amount')}
                    name="noToken"
                    type="number"
                    pattern="[0-9]*"
                    fullWidth
                    placeholder={t('inputFieldPlaceholder')}
                    handleChange={handleChange}
                    InputProps={
                      tokenName
                        ? {
                            endAdornment: (
                              <Typography
                                color="text.secondary"
                                component="span"
                                sx={endAdornmentStyles}
                              >
                                {t('noTokens')}
                              </Typography>
                            ),
                          }
                        : undefined
                    }
                    required
                  />
                </>
              )}
            </Grid>
            {expectedSwap.length > 0 && (
              <Grid item width="100%">
                <PositionSummary title={t('expectedSwap')} items={expectedSwap} />
              </Grid>
            )}
            {expectedAdjustedPosition.length > 0 && (
              <Grid item width="100%">
                <PositionSummary title={<PositionDescription />} items={expectedAdjustedPosition} />
              </Grid>
            )}
            <Grid item width="100%" flexDirection="column">
              <CustomButton
                color="primary"
                type="submit"
                label={
                  !connected
                    ? `${t('connectWallet')} + ${t(title)}`
                    : `${t(title)} for ${t(otherToken)}`
                }
                fullWidth
                disabled={!isValid}
              />
              <Typography size="body1" mt="1rem">
                {t('requiredField')}
              </Typography>
            </Grid>
          </Grid>
        </Form>
      )}
    </Formik>
  );
};
