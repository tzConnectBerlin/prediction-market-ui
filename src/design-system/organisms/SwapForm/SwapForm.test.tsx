import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { SwapForm, SwapFormProps } from './SwapForm';
import { TokenType } from '../../../interfaces';

const basicArgs = {
  tokenName: 'PMM',
  title: 'Swap Yes',
  marketId: '3',
};

const defaultArgs: SwapFormProps = {
  ...basicArgs,
  swapTokenType: TokenType.yes,
  handleSubmit: jest.fn(),
};

describe('Snapshot - render SwapForm', () => {
  it('renders correctly with default props', () => {
    const AddSwapForm = renderer.create(<SwapForm {...defaultArgs} />).toJSON();
    expect(AddSwapForm).toMatchSnapshot();
  });
});

describe('Element testing SwapForm Component', () => {
  it('render correctly SwapForm with default props', async () => {
    const { getAllByText } = render(<SwapForm {...defaultArgs} />);
    expect(getAllByText(/Swap Yes/i).length).toBe(1);
  });

  it('render correctly SwapForm with different Title', async () => {
    const { getAllByText } = render(<SwapForm {...defaultArgs} title="Swap No" />);
    expect(getAllByText(/Swap No/i).length).toBe(1);
  });
});
