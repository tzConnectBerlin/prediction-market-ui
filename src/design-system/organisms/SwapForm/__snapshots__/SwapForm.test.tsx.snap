// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot - render SwapForm renders correctly with default props 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: -24px;
  width: calc(100% + 24px);
  margin-left: -24px;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-0>.MuiGrid-item {
  max-width: none;
}

.emotion-0>.MuiGrid-item {
  padding-top: 24px;
}

.emotion-0>.MuiGrid-item {
  padding-left: 24px;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
}

.emotion-3 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 91.666667%;
  -ms-flex-preferred-size: 91.666667%;
  flex-basis: 91.666667%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 91.666667%;
}

.emotion-5 {
  color: rgba(29, 34, 39, 0.65);
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  font-family: Roboto;
  letter-spacing: 0.00938em;
  padding: 0;
  position: relative;
  display: block;
  transform-origin: top left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 133%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: translate(0, -1.5px) scale(0.75);
  -moz-transform: translate(0, -1.5px) scale(0.75);
  -ms-transform: translate(0, -1.5px) scale(0.75);
  transform: translate(0, -1.5px) scale(0.75);
  -webkit-transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  font-weight: bold;
  color: rgba(1, 102, 255, 1);
  position: relative;
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
  -webkit-transition: none;
  transition: none;
  font-size: 0.8rem;
}

.emotion-5.Mui-focused {
  color: rgba(1, 102, 255, 1);
}

.emotion-5.Mui-disabled {
  color: rgba(29, 34, 39, 0.38);
}

.emotion-5.Mui-error {
  color: rgba(255, 57, 57, 1);
}

.emotion-5.Mui-disabled {
  color: rgba(1, 102, 255, 1);
  opacity: 0.38;
}

.emotion-5 .label-asterisk {
  line-height: 0;
}

.emotion-6 {
  margin: 0;
  font-weight: 700;
  font-size: 0.75rem;
  line-height: 1.3;
  font-family: Roboto Mono;
  letter-spacing: 0.00938em;
}

.emotion-6.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 4;
}

.emotion-7.Mui-error {
  color: rgba(255, 57, 57, 1);
}

.emotion-8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  width: 100%;
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.emotion-8 input {
  text-align: inherit;
}

.emotion-9 {
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  font-family: Roboto;
  letter-spacing: 0.00938em;
  color: rgba(29, 34, 39, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  position: relative;
}

.emotion-9.Mui-disabled {
  color: rgba(29, 34, 39, 0.38);
  cursor: default;
}

label+.emotion-9 {
  margin-top: 16px;
}

.emotion-9:after {
  border-bottom: 2px solid rgba(1, 102, 255, 1);
  left: 0;
  bottom: 0;
  content: "";
  position: absolute;
  right: 0;
  -webkit-transform: scaleX(0);
  -moz-transform: scaleX(0);
  -ms-transform: scaleX(0);
  transform: scaleX(0);
  -webkit-transition: -webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  pointer-events: none;
}

.emotion-9.Mui-focused:after {
  -webkit-transform: scaleX(1);
  -moz-transform: scaleX(1);
  -ms-transform: scaleX(1);
  transform: scaleX(1);
}

.emotion-9.Mui-error:after {
  border-bottom-color: rgba(255, 57, 57, 1);
  -webkit-transform: scaleX(1);
  -moz-transform: scaleX(1);
  -ms-transform: scaleX(1);
  transform: scaleX(1);
}

.emotion-9:before {
  border-bottom: 1px solid rgba(0, 0, 0, 0.42);
  left: 0;
  bottom: 0;
  content: "\\00a0";
  position: absolute;
  right: 0;
  -webkit-transition: border-bottom-color 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: border-bottom-color 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  pointer-events: none;
}

.emotion-9:hover:not(.Mui-disabled):before {
  border-bottom: 2px solid rgba(29, 34, 39, 0.87);
}

@media (hover: none) {
  .emotion-9:hover:not(.Mui-disabled):before {
    border-bottom: 1px solid rgba(0, 0, 0, 0.42);
  }
}

.emotion-9.Mui-disabled:before {
  border-bottom-style: dotted;
}

.emotion-10 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
}

.emotion-10::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-10::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-10:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-10::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-10:focus {
  outline: 0;
}

.emotion-10:invalid {
  box-shadow: none;
}

.emotion-10::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-10.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(29, 34, 39, 0.38);
}

.emotion-10:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-11 {
  margin: 0;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.3;
  font-family: Roboto;
  letter-spacing: 0.00938em;
  color: rgba(29, 34, 39, 0.65);
  white-space: nowrap;
}

.emotion-11.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 4;
}

.emotion-12 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-13 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.8125rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 4px 10px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: rgba(255, 255, 255, 1);
  background-color: rgba(1, 102, 255, 1);
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  width: 100%;
  border-radius: 0.2em;
  padding: 0.75em 1.2em;
  border: solid 2px transparent;
  text-transform: none;
  box-shadow: none;
  color: rgba(255, 255, 255, 0.87);
}

.emotion-13::-moz-focus-inner {
  border-style: none;
}

.emotion-13.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-13 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-13:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(26, 117, 255, 1);
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-13:hover {
    background-color: rgba(1, 102, 255, 1);
  }
}

.emotion-13:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-13.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-13.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-13:hover {
  border-width: 2px;
  box-shadow: none;
}

.emotion-13:disabled {
  border-color: transparent;
}

.emotion-14 {
  margin: 0;
  font-weight: 700;
  font-size: 1rem;
  line-height: 1.3;
  font-family: Roboto Mono;
  letter-spacing: 0em;
}

.emotion-14.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 4;
}

.emotion-15 {
  margin: 0;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.3;
  font-family: Roboto;
  letter-spacing: 0.00938em;
  margin-top: 1rem;
}

.emotion-15.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 4;
}

<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column emotion-0"
  >
    <div
      className="MuiGrid-root MuiGrid-item emotion-1"
    >
      <div
        className="MuiFormControl-root emotion-2"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined emotion-3"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 emotion-4"
          >
            <label
              className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required emotion-5"
              data-shrink={true}
              htmlFor="yesToken"
            >
              <span
                className="MuiTypography-root MuiTypography-subtitle1  emotion-6"
                lines={4}
              >
                amount
              </span>
              <span
                aria-hidden={true}
                className="label-asterisk MuiInputLabel-asterisk MuiFormLabel-asterisk emotion-7"
              >
                 
                *
              </span>
            </label>
          </div>
        </div>
        <div
          alignright={false}
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root emotion-8"
          pattern="[0-9]*"
        >
          <div
            className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd emotion-9"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd emotion-10"
              disabled={false}
              name="yesToken"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="inputFieldPlaceholder"
              required={false}
              type="number"
              value=""
            />
            <span
              className="MuiTypography-root MuiTypography-body1  emotion-11"
              lines={4}
            >
              yesTokens
            </span>
          </div>
        </div>
        
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item emotion-12"
    >
      <button
        bordercolor="transparent"
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-fullWidth MuiButtonBase-root emotion-13"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        textcolor="rgba(255, 255, 255, 0.87)"
        texttype="none"
        type="button"
      >
        <h3
          className="MuiTypography-root MuiTypography-h3  emotion-14"
          lines={4}
        >
          connectWalletContinue
        </h3>
      </button>
      <p
        className="MuiTypography-root MuiTypography-body1  emotion-15"
        lines={4}
      >
        requiredField
      </p>
    </div>
  </div>
</form>
`;
